src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -427,6 +427,10 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                                 } catch (UnsupportedEncodingException e) {
                                     throw new RuntimeException("UTF not supported on this platform?", e);
                                 }
+                                if (mFailedRetries > 0) {
+                                    mFailedRetries = 0;
+                                    removeMessages(FLUSH_QUEUE);
```

Comment:
Does this remove all the FLUSH_QUEUE jobs from the work queue? What would go wrong if this line weren't here?
(This is probably related to my comment in the best-practices doc about how a single 503 can cause an otherwise quiet queue never to get flushed again. Looks like you're introducing a second instance of that same issue; not that there's necessarily anything wrong with that, for now.)